File: backend/src/controllers/seeder.controller.ts
import { Controller, Post } from '@nestjs/common';
import { SampleDataSeeder } from '../seeders/sample-data.seeder';

/**
 * کنترلر «نمونه داده» برای ایجاد داده‌های نمونه در محیط توسعه.
 */
@Controller('seeder')
export class SeederController {
  constructor(private readonly seeder: SampleDataSeeder) { }

  /** اجرای نمونه داده‌ها */
  @Post('seed')
  async seed() {
    try {
      await this.seeder.seed();
      return { message: 'Sample data seeded successfully!' };
    } catch (error) {
      return { error: 'Failed to seed sample data', details: error.message };
    }
  }
} 

File: backend-dotnet/LessonPlanner.Api/Controllers/SeederController.cs
using Microsoft.AspNetCore.Mvc;
using LessonPlanner.Api.Seeders;

namespace LessonPlanner.Api.Controllers;

[ApiController]
[Route("seeder")]
public class SeederController : ControllerBase
{
    private readonly SampleDataSeeder _seeder;

    public SeederController(SampleDataSeeder seeder)
    {
        _seeder = seeder;
    }

    [HttpPost("seed")]
    public async Task<IActionResult> Seed()
    {
        try
        {
            await _seeder.SeedAsync();
            return Ok(new { message = "Sample data seeded successfully!" });
        }
        catch (Exception ex)
        {
            return Ok(new { error = "Failed to seed sample data", details = ex.Message });
        }
    }
}
